src/main/java/com/joy/joyfillandroid/Tablefield/MobileSDKField.kt

```diff
@@ -0,0 +1,401 @@
+                TableField.TableFieldObj.RowArray = dataObjects
+                Table.ShareText.getRowTitle = tableColumnTitle
+                Table.ShareText.getRowType = tableColumnType
+                Table.ShareText.tableColumnValue = tableColumnValue
+                Table.ShareText.tableColumnOrderArrayID = tableColumnOrderArrayID
+            }
+        }
+
+    }
+
+    //Mark:- Showing rows if deleted false
+    fun filterDeletedObjects(dataArray: JSONArray): JSONArray {
+        val filteredArray = JSONArray()
+        for (i in 0 until dataArray.length()) {
+            val jsonObject = dataArray.getJSONObject(i)
+            val isDeleted = jsonObject.optBoolean("deleted", false)
+            if (isDeleted == false) {
+                filteredArray.put(jsonObject)
+            }
+        }
+
+        return filteredArray
+    }
+
+    private fun updateVaue(fields: JSONArray) {
+        for (i in 0 until fields.length()){
+            val type = fields.getJSONObject(i).get("type")
+            val id = fields.getJSONObject(i).get("_id")
+            if (type == "dropdown"){
+                val title = fields.getJSONObject(i).getString("title")
+                val options = fields.getJSONObject(i).get("options")
+                dropDown.updateDropDownvalue(title.toString(), "Options", "", options as JSONArray)
+            }
+
+            if (type == "multiSelect"){
+                val title = fields.getJSONObject(i).getString("title")
+                val options = fields.getJSONObject(i).get("options")
+                val multi = fields.getJSONObject(i).optBoolean("multi")
+                multipleChoice.setTextValue(title.toString(), options as JSONArray, multi, "fill")
+            }
+        }
+    }
+
+    private fun formatBlockData(fieldPositions: JSONArray) {
+        for (i in 0 until fieldPositions.length()){
+            val type = fieldPositions.getJSONObject(i).get("type")
+            if (type == "block"){
+                val fontSize = fieldPositions.getJSONObject(i).optInt("fontSize", 18)
+                val fontColor = fieldPositions.getJSONObject(i).optString("fontColor", "#000000")
+                val fontWeight = fieldPositions.getJSONObject(i).optString("fontWeight","normal")
+                val fontStyle = fieldPositions.getJSONObject(i).optString("fontStyle","normal")
+                val textAlign = fieldPositions.getJSONObject(i).optString("textAlign","left")
+                val i = fontSize as Int
+                val f = i.toFloat()
+                textField.setTextSize(f)
+                textField.setTextColor(Color.parseColor(fontColor.toString()))
+                setFontStyle(fontStyle as String, textField)
+                setFontWeight(fontWeight as String, textField)
+                setTextAlign(textAlign as String, textField)
+            }
+        }
+    }
+
+
+    private fun ShowComponents(shorteddatawithYPosition: List<String>, fields: JSONArray) {
```

Comment:
ShowComponents could be much better named along with the `shorteddatawithYPosition` casing is not consistent with camelCasing or others.

Reply:
done